### PR TITLE
feat: BTD-268 env variable MAPPINGS can point to a folder now

### PR DESCRIPTION
**Pull request by evialle73**
Task BTD-268 from Epic BTD-272

The env variable MAPPINGS can point to a folder now:

- If MAPPINGS Point to a folder: all json files in this folder are loaded and aggregated by the tool internally
- If MAPPINGS Point to a single json file: the file is loaded and expected to be an aggregation of mapping files (old behavior)

I also added a `./mapping` folder in the project to perform some tests locally.

### CODE REVIEW

File: mappings/bridge-alps.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
```
Comment (aureliencousin) [inline mappings/bridge-alps.json:1]: If those files are meant for testing purposes, please make this explicit (like moving them to a test/fixtures/ folder)

File: mappings/bridge-alps.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  "leads": {
+    "pk": "id",
+    "_id": {
```
Comment (aureliencousin) [inline mappings/bridge-alps.json:4]: If those files are meant for testing purposes, we don't need all of them. If the goal is to be able to test the aggregation, then 2 mapping files should be enough

**Review by aureliencousin — CHANGES_REQUESTED** [review]
The pull request only contains the json files, the actual changed code is missing

File: mappings/bridge-alps.json (hunk 2 above)
Comment (evialle73) [reply mappings/bridge-alps.json:4]: They are not the same, I want all of them for later (mongodump import)

File: mappings/bridge-alps.json (hunk 1 above)
Comment (evialle73) [reply mappings/bridge-alps.json:1]: yes, I will do that

Comment (evialle73) [reply mappings/bridge-alps.json]: done

File: mappings/bridge-alps.json (hunk 2 above)
Comment (aureliencousin) [reply mappings/bridge-alps.json]: ok

File: src/main/java/com/malt/mongopostgresqlstreamer/configurations/MongoConfiguration.java [hunk 3]
```diff
@@ -1,36 +1,37 @@
...
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientURI;
+import com.mongodb.client.MongoDatabase;
+
 @Configuration
 public class MongoConfiguration {
 
-    @Value(value = "${mongo.uri:mongodb://localhost:27017}")
-    private String uri;
+	@Value(value = "${mongo.uri:mongodb://localhost:27017}")
```
Comment (jeremybviseo) [inline src/main/java/com/malt/mongopostgresqlstreamer/configurations/MongoConfiguration.java:15]: issue with IDE vs original project incrementation settings

**Review by aureliencousin — APPROVED** [review]

File: test/mappings/bridge-alps.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{
```
Comment (jeremybviseo) [inline test/mappings/bridge-alps.json:1]: 🤔 why does mapping files commited ?

File: src/main/java/com/malt/mongopostgresqlstreamer/configurations/MongoConfiguration.java (hunk 3 above)
Comment (evialle73) [reply src/main/java/com/malt/mongopostgresqlstreamer/configurations/MongoConfiguration.java:15]: I plan to do a major cleanup later, is it ok for you ?

**Comment by aureliencousin** [comment]
code looks good to me regarding current project quality.

File: test/mappings/bridge-alps.json (hunk 4 above)
Comment (evialle73) [reply test/mappings/bridge-alps.json:1]: those are for testing purpose only. They can be outdated, that's not an issue. They will be important later to test the mongodump import

File: src/main/java/com/malt/mongopostgresqlstreamer/configurations/MongoConfiguration.java (hunk 3 above)
Comment (jeremybviseo) [reply src/main/java/com/malt/mongopostgresqlstreamer/configurations/MongoConfiguration.java:15]: ok

File: test/mappings/bridge-alps.json (hunk 4 above)
Comment (jeremybviseo) [reply test/mappings/bridge-alps.json:1]: ok

File: src/main/java/com/malt/mongopostgresqlstreamer/StreamerApplication.java [hunk 5]
```diff
@@ -1,81 +1,118 @@
...
+			throw new RuntimeException("Mongodump input folder not set");
+		}
+		if (!inputFolder.exists()) {
+			log.error("Mongodump input folder does not exists: " + inputFolder.getAbsolutePath());
+		}
+		List<File> dumps = Arrays.stream(inputFolder.listFiles())
+				.filter(File::isFile)
+				.filter(f -> f.getName()
+						.endsWith(".gz"))
+				.collect(Collectors.toList());
```
Comment (aureliencousin) [inline src/main/java/com/malt/mongopostgresqlstreamer/StreamerApplication.java:66]: please remove this unfinished function from this PR (it's not relevant regarding commit description)

**Review by aureliencousin — CHANGES_REQUESTED** [review]

**Pull request closed without merging**